flow/workflows/cdc_flow.go

```diff
@@ -519,16 +521,26 @@ func CDCFlowWorkflow(
 
 	originalRunID := workflow.GetInfo(ctx).OriginalRunID
 
-	var err error
-	ctx, err = GetFlowMetadataContext(ctx, &protos.FlowContextMetadataInput{
-		FlowName:        cfg.FlowJobName,
-		SourceName:      cfg.SourceName,
-		DestinationName: cfg.DestinationName,
-		Status:          state.CurrentFlowStatus,
-		IsResync:        cfg.Resync,
-	})
-	if err != nil {
-		return state, fmt.Errorf("failed to get flow metadata context: %w", err)
+	for {
+		if err := ctx.Err(); err != nil {
+			state.updateStatus(ctx, logger, protos.FlowStatus_STATUS_TERMINATED)
+			return state, err
+		}
+
+		var err error
+		ctx, err = GetFlowMetadataContext(ctx, &protos.FlowContextMetadataInput{
+			FlowName:        cfg.FlowJobName,
+			SourceName:      cfg.SourceName,
+			DestinationName: cfg.DestinationName,
+			Status:          state.CurrentFlowStatus,
+			IsResync:        cfg.Resync,
+		})
+		if err != nil {
+			logger.Error("failed to GetFlowMetadataContext", slog.Any("error", err))
+			continue
```

Comment:
we should probably add a sleep here to prevent overloading catalog

Reply:
tbh this will rarely happen because temporal will be retrying the local activity in this scenario, `Get`'s failure would probably be rooted in type mismatch which we can't really do anything about besides not have type errors